js/src/name/publish.js

```diff
@@ -0,0 +1,96 @@
+/* eslint-env mocha */
+'use strict'
+
+const each = require('async/each')
+const hat = require('hat')
+
+const { fixtures } = require('./utils')
+const { getDescribe, getIt, expect } = require('../utils/mocha')
+
+module.exports = (createCommon, options) => {
+  const describe = getDescribe(options)
+  const it = getIt(options)
+  const common = createCommon()
+
+  describe('.name.publish', function () {
+    this.timeout(50 * 1000)
+
+    const keyName = hat()
+    let ipfs
+    let nodeId
+    let keyId
+
+    before(function (done) {
+      this.timeout(60 * 1000)
+
+      common.setup((err, factory) => {
+        expect(err).to.not.exist()
+
+        factory.spawnNode((err, node) => {
+          expect(err).to.not.exist()
+
+          ipfs = node
+          ipfs.id().then((res) => {
+            expect(res.id).to.exist()
+
+            nodeId = res.id
+
+            ipfs.key.gen(keyName, { type: 'rsa', size: 2048 }, (err, key) => {
+              expect(err).to.not.exist()
+              expect(key).to.exist()
+              expect(key).to.have.property('name', keyName)
+              expect(key).to.have.property('id')
+
+              keyId = key.id
+              populate()
+            })
+          })
+        })
+      })
+
+      function populate () {
+        each(fixtures.files, (file, cb) => {
+          ipfs.files.add(file.data, { pin: false }, cb)
+        }, done)
+      }
```

Comment:
Please move this into a separate `before` block (they are executed in series in the order they appear) so the setup is a bit easier to read.

Reply:
With the `before` refactor, and only adding a single file now, I think that it will be better to have all together. When I submit a new commit, let me know if you agree 🙂